test/test_config.py

```diff
@@ -198,8 +199,11 @@ def test_makedirs(temp_dir):
     d = os.path.join(temp_dir.strpath, 'foo')
     assert os.path.isdir(d)
 
-    expected = (7 * 64 + 5 * 8 + 5)  # Octal 755
-    assert expected == (os.lstat(d).st_mode & 0o777)
+    curmask = os.umask(0)
+    os.umask(curmask)
+
+    expected = 0o777 & ~curmask
```

Comment:
I only tested this on Linux, but I did adjust the umask before running to verify it worked for any umask I set as well.